protocol/river-layout-unstable-v1.xml

```diff
@@ -0,0 +1,218 @@
+
+        All layout related communication is done through this interface.
+
+        The namespace is used by the compositor to decide which river_layout
+        object will receive layout demands for the output. Namespaces must be
+        globally unique. If the namespace is already in use by a different
+        river_layout object, the server will send a namespace_in_use event on
+        the river_layout object immediately after creation and ignore all
+        requests on the object aside from the destroy request.
+      </description>
+      <arg name="id" type="new_id" interface="zriver_layout_v1"/>
+      <arg name="output" type="object" interface="wl_output"/>
+      <arg name="namespace" type="string" summary="namespace of the layout object"/>
+    </request>
+  </interface>
+
+  <interface name="zriver_layout_v1" version="1">
+    <description summary="receive and respond to layout demands">
+      This interface allows clients to receive layout demands from the
+      compositor for a specific output and subsequently propose positions and
+      dimensions of individual views.
+    </description>
+
+    <enum name="error">
+      <entry name="invalid_serial" value="0" summary="serial does not belong to an active layout demand"/>
+      <entry name="invalid_index" value="1" summary="index does not belong to a view in the list"/>
+      <entry name="missing_index" value="2" summary="one or multiple view indices have not been used in the layout"/>
+    </enum>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the river_layout object">
+        This request indicates that the client will not use the river_layout
+        object any more.
+      </description>
+    </request>
+
+    <event name="namespace_in_use">
+      <description summary="namespace already in use">
+        The requested namespace is already used by another river_layout object.
+        After raising this event, the server will ignore all request on the
+        river_layout object aside from the destroy request.
+      </description>
+    </event>
+
+    <event name="layout_demand">
+      <description summary="the compositor is in demand of a layout">
+        The compositor raises this event to inform the client that it requires a
+        layout for a set amount of views.
+
+        The useable width and height height indicate the space in which the
+        client can safely position views without interfering with desktop
+        widgets such as panels.
+
+        Note that the main area related values are simply proposals and can
+        safely be ignored by the client.
+
+        The serial of this event is used to identify subsequent events and
+        request as belonging to this layout demand. Beware that the client
+        might need to handle multiple layout demands at the same time.
+
+        A client does not need to respond to this event.
```

Comment:
I think this should be clarified as good clients will respond to every event unless preemption occurs.
```suggestion
        The server will ignore responses to all but the most recent layout command. Thus, clients should only respond to the most recent layout_demand received. If a newer layout_demand is received before the client has finished responding to an old demand, the client should abort work on the old demand as any further work would be wasted.
```